pandas/core/frame.py

```diff
@@ -7023,6 +7023,8 @@ def append(
         """
         if isinstance(other, (Series, dict)):
             if isinstance(other, dict):
+                if not ignore_index:
```

Comment:
Rather than adding another branch here can you just use an f-string and do `f"Can only append a {type(other)}...`

Reply:
Could do, but then I'll need to add another branch later as the message for Series has an extra part compared to that for dict: "or if the Series has a name"